### DISCUSSION TITLE
point cloud precision

### DISCUSSION THREAD

**Discussion by mikelondon001** [isl-org/Open3D#4436] [Q&A]
I'm displaying a point cloud with open3d.geometry.PointCloud(). The coordinates are UTM georeferenced, meaning that the north-south coordinates (where I am) are all in the vicinity of 6.3e6. The display shows symptoms of limited precision - points that should be uniformly distributed fall on a few horizontal lines. My coordinates are close to the limit of 32 bit floating point precision, and I wonder if this is what I'm seeing. I'm hoping somebody with knowledge of the internals could confirm/deny this or point me to a relevant discussion.
Thanks.

**Comment by mikelondon001** [comment]
I forgot to mention that when I display clouds relative to some local reference point - i.e. subtract some large number from all the coordinates - everything looks fine.

**Comment by errissa** [comment]
@mikelondon001 Yes, you're seeing floating point precision problems and as you've already noted one solution to the problem is to transform the scene into the coordinate system of the camera. The new rendering backend does that automatically and you should not have precision problems. As a quick test you can pass your point cloud into the `open3d.visualization.draw(my_point_cloud)`.
